doc/glossary.rst

```diff
@@ -0,0 +1,60 @@
+Glossary
+========
+
+Expressions that are grouped together are used synonymously in the Strawberry Fields documentation.
+
+
+.. glossary::
+
+   backend
+      Executes quantum :term:`programs <program>`. Can be either a classical simulator or a
+      quantum hardware device, local or remote.
+      See: :class:`.BaseBackend`
+
+   circuit
+   quantum circuit
+   program
+      A sequence of quantum :term:`operations <operation>` acting on a quantum :term:`register`.
+      Typically contains :term:`measurements <measurement>` that produce a stochastic classical output.
+      Represents a quantum computation.
+      See: :class:`.Program`
+
+   compilation
+      The process of converting a quantum :term:`program` into an :term:`equivalent program` that
+      can be run on a specific :term:`device`. The compilation can fail if the device is incapable
+      of running the program.
+
+   device
```

Comment:
@smite @josh146 the name "device" is definitely the most unclear concept. Suggest choosing better name

Possibilities:
1. circuit family
1. circuit type
1. circuit class
1. program spec
1. subprogram
1. program family
1. program type
1. program class

Reply:
Of the above, I think I like 'circuit class' or 'circuit type' the most, but it still doesn't sound quite right...

Reply:
I guess "circuit template" is already taken, and means a parametrized circuit?